tests_v2/test_read_selection.py

```
@@ -0,0 +1,148 @@
+            # 2) CLI without selection: non-zero exit + explicit error.
+            no_sel = subprocess.run(
+                [cli, "--socket", SOCKET_PATH, "read-screen", "--workspace", ws, "--surface", sf, "--selection"],
+                capture_output=True,
+                text=True,
+                check=False,
+            )
+            _must(no_sel.returncode != 0, "Expected read-screen --selection to fail without a selection")
+            _must("no active selection" in f"{no_sel.stdout}\n{no_sel.stderr}", f"Unexpected no-selection error: {no_sel.stderr!r}")
+
+            # 3) --selection is incompatible with --scrollback/--lines.
+            bad_combo = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)